tools/src/test/java/org/neo4j/tools/applytx/InteractiveApplyTransactionsTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.tools.applytx;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.InputStream;
+
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.graphdb.factory.GraphDatabaseFactory;
+import org.neo4j.test.DbRepresentation;
+import org.neo4j.test.TargetDirectory;
+
+import static org.junit.Assert.assertEquals;
+
+import static java.lang.String.format;
+
+public class InteractiveApplyTransactionsTest
```

Comment:
I would love to see some tests for the interactive features as well.

It obviously cannot run in an interactive loop (at least that would be too bothersome), but we could probably expose a method for passing a command string in, and evaluating it, then use that to unit test each `Action`.